Packs/Rapid7_Nexpose/Integrations/Rapid7_Nexpose/Rapid7_Nexpose.py

```diff
@@ -4928,7 +4928,22 @@ def start_site_scan_command(client: Client, site_id: str | None = None, site_nam
 
     else:
         assets = client.get_site_assets(site.id)
-        hosts_list = [asset["ip"] for asset in assets]
+
+        hosts_list = set()  # Using a set to avoid duplicates
+
+        for asset in assets:
+            if asset.get("ip"):
+                hosts_list.add(asset["ip"])
+
+            # In some cases there is an IP address in the "addresses" field, but not in the "ip" field.
+            elif asset.get("addresses"):
+                for address in asset["addresses"]:
+                    if address.get("ip"):
+                        hosts_list.add(address["ip"])
+
+        hosts_list = list(hosts_list)
+        # Using sort to allow consistent unit-testing. Seems like set -> list conversion results in a random order.
+        hosts_list.sort()
```

Comment:
if we need this `sort` only for testing, so do it in the test.
No need to add time complexity to the code because of unit testing.

Reply:
Fixed be236d36ce6e82f49e52312bf3fcd63912e98170